remix-simulator/src/methods/accounts.js

```diff
@@ -1,27 +1,66 @@
+var RemixLib = require('remix-lib')
+var executionContext = RemixLib.execution.executionContext
+var ethJSUtil = require('ethereumjs-util')
+var BN = ethJSUtil.BN
 var Web3 = require('web3')
 
 var Accounts = function () {
   this.web3 = new Web3()
   // TODO: make it random and/or use remix-libs
-  this.accounts = [this.web3.eth.accounts.create(['abcd']), this.web3.eth.accounts.create(['ef12']), this.web3.eth.accounts.create(['ef34'])]
+  this.accountsList = [this.web3.eth.accounts.create(['abcd']), this.web3.eth.accounts.create(['ef12']), this.web3.eth.accounts.create(['ef34'])]
+  this.accounts = {}
+  this.accountsKeys = {}
 
-  this.accounts[this.accounts[0].address.toLowerCase()] = this.accounts[0]
-  this.accounts[this.accounts[1].address.toLowerCase()] = this.accounts[1]
-  this.accounts[this.accounts[2].address.toLowerCase()] = this.accounts[2]
+  executionContext.init({get: () => { return true }})
 
-  this.accounts[this.accounts[0].address.toLowerCase()].privateKey = Buffer.from(this.accounts[this.accounts[0].address.toLowerCase()].privateKey.slice(2), 'hex')
-  this.accounts[this.accounts[1].address.toLowerCase()].privateKey = Buffer.from(this.accounts[this.accounts[1].address.toLowerCase()].privateKey.slice(2), 'hex')
-  this.accounts[this.accounts[2].address.toLowerCase()].privateKey = Buffer.from(this.accounts[this.accounts[2].address.toLowerCase()].privateKey.slice(2), 'hex')
+  for (let _account of this.accountsList) {
+    this.accountsKeys[_account.address.toLowerCase()] = _account.privateKey
+    this.accounts[_account.address.toLowerCase()] = { privateKey: Buffer.from(_account.privateKey.replace('0x', ''), 'hex'), nonce: 0 }
+
+    executionContext.vm().stateManager.getAccount(Buffer.from(_account.address.toLowerCase().replace('0x', ''), 'hex'), (err, account) => {
+      if (err) {
+        throw new Error(err)
+      }
+      var balance = '0x56BC75E2D63100000'
+      account.balance = balance || '0xf00000000000000001'
```

Comment:
is that enough for adding balance here?
I think we might need to do something like https://github.com/ethereum/remix-ide/blob/swap_it/src/universal-dapp.js#L127

Reply:
The api changed afak

Reply:
but this call `stateManager.putAccount` is still needed no?

Reply:
turns out it's not needed anymore. getAccount will return the account object which can be modified. a way to verify this is to run a script like

```
const Web3 = require('web3')
const RemixSim = require('./')

provider = new RemixSim.Provider({})
web3 = new Web3(provider)

web3.eth.getAccounts((error, accounts) => {
  let account = accounts[0]

  web3.eth.getBalance(account).then((balance) => {
    console.dir("=============")
    console.dir(balance)
    console.dir("=============")
  })
});
```

`balance` will have the value of `var balance = '0x56BC75E2D63100000'` changing this will the balance returned above